src/Identity/Extensions.Core/src/UserClaimsPrincipalFactory.cs

```diff
@@ -76,11 +76,17 @@ protected virtual async Task<ClaimsIdentity> GenerateClaimsAsync(TUser user)
         {
             var userId = await UserManager.GetUserIdAsync(user);
             var userName = await UserManager.GetUserNameAsync(user);
+            var email = await UserManager.GetEmailAsync(user);
```

Comment:
You probably have to check whether email is supported before doing this, similar to the claims stamp check below

Reply:
@Haok Done! Do I need to check weather email claim is null or empty before adding it as a claim?

Reply:
It doesn't hurt so why not